### ISSUE TITLE
[Add2App iOS] Frames are still being processed when LifeCycle state is inactive, after closing a Flutter view and reusing the same Flutter Engine  for subsequent views

### ISSUE THREAD

**Issue by dannyvalentesonos** [flutter/flutter#45595]
This is a rather complicated repro, but causes some bad behavior when it happens.

On iOS, after creating a Flutter Engine and a FlutterViewController using the created engine, any time the Control Center is pulled up from the bottom, `_handleBeginFrame` in `scheduler/binding.dart` is not called anymore while the Control Center is showing. `_handleBeginFrame` resumes getting called once the Control Center is closed.

However, once the FlutterViewController is destroyed, the next time a new FlutterViewController is created, using the same Engine as before, any time the Control Center is pulled up, `_handleBeginFrame` continues getting called. This causes animations to run without ever being rendered while the Control Center is showing.

I'm not sure what the actual correct behavior is supposed to be... Should frames still be scheduled while the Control Center is showing, or not? But the inconsistency is causing us problems.

**Comment by dannyvalentesonos** [comment]
@dnfield FYI.

**Comment by dnfield** [comment]
@xster or @gaaclarke may have thoughts about this

**Comment by dannyvalentesonos** [comment]
Here's the output of flutter doctor.
```
[✓] Flutter (Channel dev, v1.10.14, on Mac OS X 10.14.6 18G103, locale en-US)
    • Flutter version 1.10.14 at /Users/danny.valente/Library/flutter/1.10.14-dev
    • Framework revision 1946fc4da0 (7 weeks ago), 2019-10-07 15:23:31 -0700
    • Engine revision 1d62160fdb
    • Dart version 2.6.0 (build 2.6.0-dev.1.0 d6c6d12ebf)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/danny.valente/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
```

**Comment by xster** [comment]
Sounds indeed like an issue. Perhaps a dangling nsnotificationcenter observer in the engine. We should add tests for these cases.

**Comment by xster** [comment]
@dnfield do you have the bandwidth to take a swing at this?

**Comment by dnfield** [comment]
Not until later this month at the earliest.  If someone else hasn't gotten to it by then I can.

**Comment by iapicca** [comment]
Hi @dannyvalentesonos 
if the issue persist with the latest stable version of flutter
could you please provide your updated `flutter doctor -v` ,
Thank you

**Comment by dannyvalentesonos** [comment]
Yes it does happen with the latest stable.

```
[✓] Flutter (Channel beta, v1.12.13+hotfix.6, on Mac OS X 10.14.6 18G3020, locale en-US)
    • Flutter version 1.12.13+hotfix.6 at /Users/danny.valente/Library/flutter/1.12.13+hotfix.6-beta
    • Framework revision 18cd7a3601 (3 months ago), 2019-12-11 06:35:39 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/danny.valente/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
```

**Comment by TahaTesser** [comment]
Hi @dannyvalentesonos 
Can you please provide a complete reproducible minimal code sample and steps to reproduce 
Thank you

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by dannyvalentesonos** [comment]
@TahaTesser I'm very sorry for the long wait. I had to create a sample Add2App that demonstrates the problem.

I'm attaching the sample project, and also a video showing the issue.

Steps to repro:
1. Launch app
2. Launch the flutter view from the native app using the "Launch Flutter!" button
3. Press the "Move" button, and then pull up the Control Center.
4. Wait 5 seconds and notice no "completed" output shows.
5. Close the Control Center, and notice the "completed" output immediately shows, because the frames are processed now and the animation duration is overdue.
6. Select the "Close" button.
7. Repeat steps 2 - 4, but now notice the "completed" output shows while the Control Center is still showing.

This inconsistency is the problem.

[Add2AppTest_background_animations.zip](https://github.com/flutter/flutter/files/5643474/Add2AppTest_background_animations.zip)
[background_animation_problem.mov.zip](https://github.com/flutter/flutter/files/5643475/background_animation_problem.mov.zip)


Can we please get this bug reopened? Thanks!

**Comment by TahaTesser** [comment]
Hi @dannyvalentesonos 
Thanks for reproducible code sample! I can reproduce it 
<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.25.0-5.0.pre.92, on macOS 11.0.1 20B29
    darwin-x64, locale en-GB)
    • Flutter version 1.25.0-5.0.pre.92 at /Users/tahatesser/Code/flutter_master
    • Framework revision 90f08769a3 (2 hours ago), 2020-12-07 12:33:03 +0100
    • Engine revision 6491c7518f
    • Dart version 2.12.0 (build 2.12.0-120.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (4 available)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E            • ios
      • iOS 14.2
    • iPhone 12 (mobile)   • C303B5A5-D598-4CAC-8ACC-09BF07A191DC • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-2 (simulator)
    • macOS (desktop)      • macos                                • darwin-x64
      • macOS 11.0.1 20B29 darwin-x64
    • Chrome (web)         • chrome                               •
      web-javascript • Google Chrome 87.0.4280.88

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
I tried to verify this issue to see if the reported behavior still persists using latest versions or not, but it seems the sample project provided above is not complete, as the `ios` folder doesn't have `.xcworkspace` file in order to run the add2app scenario, so I am unable to verify this issue.

@dannyvalentesonos Is there any chance that you could provide an updated project sample if the issue is still replicating at your end ? Considering age of the issue and lack of any activity after last update on this issue, we first would like to confirm the issue status before we could update this issue.

**Comment by dannyvalentesonos** [comment]
[Add2AppTest_background_animations.zip](https://github.com/flutter/flutter/files/10558030/Add2AppTest_background_animations.zip)

**Comment by darshankawar** [comment]
Thanks for the prompt response and providing further details @dannyvalentesonos 
I verified using the updated information and was unable to replicate the reported behavior. Upon following steps to replicate, I see below in the console log:

```
2023-02-07 16:57:59.382825+0530 Add2AppTest[35679:316576] flutter: animationg status changed to AnimationStatus.forward
2023-02-07 16:58:04.376377+0530 Add2AppTest[35679:316576] flutter: animationg status changed to AnimationStatus.completed
```

With this and also based on your last comment, I am going ahead and closing this as fixed.
